src/pages/index.js

```diff
@@ -7,36 +7,86 @@ import UserInfo from "../components/UserInfo.js";
 import { initialCards } from "../utils/constants.js";
 import { config } from "../utils/constants.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
+import Popup from "../components/Popup.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
+
+let section;
+
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "953ea6cd-2c01-4d7b-be00-e077aa224921",
+    "Content-Type": "application/json",
+  },
+});
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    setIsLiked,
+    handleDelete
+  );
   const cardElement = card.getCardView();
+
   return cardElement;
 }
 
+function setIsLiked(card) {
+  if (card.isLiked()) {
+    api.dislikeCard(card.getCardId()).then((res) => {
+      card.setIsLiked(false),
+        (err) => {
+          console.error(err);
+          alert(`${err}, something went wrong`);
+        };
+    });
+  } else {
+    api.likeCard(card.getCardId()).then(
+      (res) => card.setIsLiked(true),
+      (err) => {
```

Comment:
Needs correction: the errors should be handled in the catch block:

```js
 api.dislikeCard(card.getCardId()).then((res) => {
  card.setIsLiked(false)
}).catch((err) => {
  // handle the error
})
```

Reply:
<img width="791" alt="Screenshot 2024-06-06 at 10 31 14 PM" src="https://github.com/alphaxuzi/se_project_aroundtheus/assets/109047750/051ca073-53d4-419a-9e22-fdb07ae29525">

Reply:
Yes, that is because you called catch after setIsLiked, but you should do that in the api promise chain